controllers/sensor/validate.go

```diff
@@ -39,6 +40,27 @@ func ValidateSensor(s *v1alpha1.Sensor) error {
 	if len(s.Spec.DeploySpec.Containers) > 1 {
 		return fmt.Errorf("sensor pod specification can't have more than one container")
 	}
+	switch s.Spec.EventProtocol.Type {
+	case pc.HTTP:
+		if s.Spec.EventProtocol.Http.Port == "" {
+			return fmt.Errorf("http server port is not defined")
+		}
+	case pc.NATS:
+		if s.Spec.EventProtocol.Nats.URL == "" {
+			return fmt.Errorf("nats url is not defined")
+		}
+		if s.Spec.EventProtocol.Nats.Type == "" {
+			return fmt.Errorf("nats type is not defined. either Standard or Streaming type should be defined")
+		}
+		if s.Spec.EventProtocol.Nats.Type == pc.Streaming && s.Spec.EventProtocol.Nats.ClientId == "" {
```

Comment:
this is okay, but repeating `if` statements. consider nesting of conditions to simplify these